com.unity.render-pipelines.core/Editor/Volume/VolumeComponentEditor.cs

```diff
@@ -202,37 +212,27 @@ public override int Compare((GUIContent displayName, int displayOrder, Serialize
         /// </remarks>
         public virtual void OnEnable()
         {
-            m_Parameters = new List<(GUIContent, int, SerializedDataParameter)>();
-
             // Grab all valid serializable field on the VolumeComponent
             // TODO: Should only be done when needed / on demand as this can potentially be wasted CPU when a custom editor is in use
-            var fields = target.GetType()
-                .GetFields(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
-                .Where(t => t.FieldType.IsSubclassOf(typeof(VolumeParameter)))
-                .Where(t =>
-                    (t.IsPublic && t.GetCustomAttributes(typeof(NonSerializedAttribute), false).Length == 0) ||
-                    (t.GetCustomAttributes(typeof(SerializeField), false).Length > 0)
-                    )
-                .Where(t => t.GetCustomAttributes(typeof(HideInInspector), false).Length == 0)
-                .ToList();
-
-            // Prepare all serialized objects for this editor
-            foreach (var field in fields)
-            {
-                var property = serializedObject.FindProperty(field.Name);
-                var name = "";
-                var order = 0;
-                var attr = (DisplayInfoAttribute[])field.GetCustomAttributes(typeof(DisplayInfoAttribute), true);
-                if (attr.Length != 0)
-                {
-                    name = attr[0].name;
-                    order = attr[0].order;
-                }
+            var fields = new List<(FieldInfo, SerializedProperty)>();
+            GetFields(target, fields);
+
+            m_Parameters = fields
+                .Select(t => {
+                    var name = "";
+                    var order = 0;
+                    var attr = (DisplayInfoAttribute[])t.Item1.GetCustomAttributes(typeof(DisplayInfoAttribute), true);
+                    if (attr.Length != 0)
+                    {
+                        name = attr[0].name;
+                        order = attr[0].order;
+                    }
 
-                var parameter = new SerializedDataParameter(property);
-                m_Parameters.Add((new GUIContent(name), order, parameter));
-            }
-            m_Parameters.Sort(new ParameterSorter());
+                    var parameter = new SerializedDataParameter(t.Item2);
+                    return (new GUIContent(name), order, parameter);
```

Comment:
If I get it well, here will be the field name right? Shouldn't it be localizable?

Reply:
yeah, that's true, I will create a another task to improve this. Thanks Rémi